src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,80 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add shopping cart to DB: " + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.id = :id", ShoppingCart.class);
+            query.setParameter("id", user.getId());
+            ShoppingCart shoppingCart = query.getSingleResult();
+            for (Ticket ticket : shoppingCart.getTickets()) {
+                Hibernate.initialize(ticket.getMovieSession().getMovie());
+                Hibernate.initialize(ticket.getMovieSession().getCinemaHall());
+            }
```

Comment:
```suggestion
            }
```
add fetch joins for movie and cinema hall m to avoid additional requests to a db